jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SniX509ExtendedKeyManager.java

```diff
@@ -81,29 +88,51 @@ protected String chooseServerAlias(String keyType, Principal[] issuers, Collecti
                 {
                     SslContextFactory.AliasSNIMatcher matcher = (SslContextFactory.AliasSNIMatcher)m;
                     host = matcher.getHost();
-                    x509 = matcher.getX509();
                     break;
```

Comment:
I know this code is unchanged, but I don't like the look of the loop looking for just the first matcher and using that.  We should at least have a comment saying "find our matcher" to indicate we know their should only be one.   Perhaps use the streams API with a findFirst to make it clear.

We should probably also explain that our SniMatcher always "match" even if there is no SNI or not a match and that the presence of a host field tells us there was an SNI in the connection and the presence of x509 tells us there was really a name match.  The comment in our AliasSniMatcher says:
``` java
            // Return true and allow the KeyManager to accept or reject when choosing a certificate.
            // If we don't have a SNI host, or didn't see any certificate aliases,
            // just say true as it will either somehow work or fail elsewhere.
```